eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/MessageOnion.scala

```diff
@@ -118,17 +119,19 @@ object MessageOnionCodecs {
     .typecase(UInt64(68), variableSizeBytesLong(varintoverflow, invoiceErrorCodec.as[InvoiceError]))
 
 
-  val perHopPayloadCodec: Codec[TlvStream[OnionMessagePayloadTlv]] = TlvCodecs.lengthPrefixedTlvStream[OnionMessagePayloadTlv](onionTlvCodec).complete
+  val perHopPayloadCodec: Codec[TlvStream[OnionMessagePayloadTlv]] = TlvCodecs.tlvStream[OnionMessagePayloadTlv](onionTlvCodec).complete
 
-  val relayPerHopPayloadCodec: Codec[RelayPayload] = perHopPayloadCodec.narrow({
+  val prefixedPerHopPayloadCodec: Codec[TlvStream[OnionMessagePayloadTlv]] = variableSizeBytesLong(CommonCodecs.varintoverflow, perHopPayloadCodec)
```

Comment:
This change seems unnecessary, you're not using the non-length-prefixed `perHopPayloadCodec` anywhere?

Reply:
Removed